front/components/vaults/EditVaultManagedDatasourcesViews.tsx

```diff
@@ -197,15 +199,32 @@ export function EditVaultManagedDataSourcesViews({
         }}
         initialSelectedDataSources={vaultDataSourceViews}
       />
-
+      <Dialog
```

Comment:
NIT: Maybe we can render the VaultManagedDataSourcesViewsModal & this Dialog only isAdmin?